packages/rocket-static-sites-aws-infrastructure/src/static-website-stack.ts

```diff
@@ -0,0 +1,76 @@
+import { CfnOutput, RemovalPolicy, Stack } from '@aws-cdk/core'
+import { Bucket } from '@aws-cdk/aws-s3'
+import { BucketDeployment, Source } from '@aws-cdk/aws-s3-deployment'
+import {
+  CloudFrontWebDistribution,
+  OriginAccessIdentity,
+  ViewerCertificate,
+  ViewerProtocolPolicy,
+} from '@aws-cdk/aws-cloudfront'
+import { existsSync } from 'fs'
+import { RocketUtils } from '@boostercloud/framework-provider-aws-infrastructure'
+
+export type AWSStaticSiteParams = {
+  localDistPath?: string
+  indexFile?: string
+  errorFile?: string
+  bucketName: string
+}
+
+export class StaticWebsiteStack {
+  public static mountStack(params: AWSStaticSiteParams, stack: Stack): void {
+    const localDistPath = params.localDistPath ?? './public'
+    const indexFile = params.indexFile ?? 'index.html'
+    const errorFile = params.errorFile ?? '404.html'
+    if (existsSync(localDistPath)) {
+      const staticWebsiteOIA = new OriginAccessIdentity(stack, 'staticWebsiteOIA', {
+        comment: 'Allows static site to be reached only via CloudFront',
+      })
+      const staticSiteBucket = new Bucket(stack, 'staticWebsiteBucket', {
+        websiteIndexDocument: indexFile,
+        websiteErrorDocument: errorFile,
+        bucketName: params.bucketName,
```

Comment:
I see the `bucketName` does not have a default name, but I think we could provide one and the user would not need to specify it.

What do you think about using `config.applicationName` + `static-site-bucket`?

Reply:
This would require many changes because in the current implementation we don't have access to `config`. I've created issue #482 to tackle this.